src/contexts/favBlog/FavBlogContext.js

```diff
@@ -0,0 +1,58 @@
+import React, { createContext, useContext, useReducer } from "react";
+
+// create context
+const FavBlogPostsContext = createContext();
+
+//The reducer function contains your custom state logic
+//and the initialState can be a simple value but generally will contain an object.
+const initialState = { favBlogs: [] };
+
+function favBlogReducer(state, action) {
+  const { type, payload } = action;
+  const { favBlogs } = state;
+  switch (type) {
+    case "ADD_FAV_BLOG":
+      if (!favBlogs.find((blog) => blog.id === payload.id)) {
```

Comment:
Try to be explicit in the conditions:
```suggestion
const favoriteBlogExists = favBlogs.find((blog) => blog.id === payload.id);
      if (!favoriteBlogExists) {
```

Reply:
Great point. Add clarity.